### PR TITLE
#383: Add security scan for flavors

### PR DESCRIPTION
**Pull request by tomuben**

### CODE REVIEW

File: flavors/python-3.6-data-science-EXASOL-6.2.0/flavor_base/security_scan/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM {{release}}
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN mkdir -p /build_info/packages
+COPY security_scan/packages /build_info/packages/security_scan
+RUN /scripts/install_scripts/install_via_apt.pl --file /build_info/packages/security_scan/apt_get_packages_trivy_deps
+RUN wget -qO - https://aquasecurity.github.io/trivy-repo/deb/public.key | apt-key add - &&\
```
Comment (tkilias) [inline flavors/python-3.6-data-science-EXASOL-6.2.0/flavor_base/security_scan/Dockerfile:7]:
```suggestion
RUN wget -qO - https://aquasecurity.github.io/trivy-repo/deb/public.key | apt-key add - && \
```

Comment (tkilias) [reply flavors/python-3.6-data-science-EXASOL-6.2.0/flavor_base/security_scan/Dockerfile:7]: for all others well

File: flavors/python-3.6-data-science-EXASOL-6.2.0/flavor_base/security_scan/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM {{release}}
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN mkdir -p /build_info/packages
+COPY security_scan/packages /build_info/packages/security_scan
+RUN /scripts/install_scripts/install_via_apt.pl --file /build_info/packages/security_scan/apt_get_packages_trivy_deps
+RUN wget -qO - https://aquasecurity.github.io/trivy-repo/deb/public.key | apt-key add - && \
```
Comment (tkilias) [inline flavors/python-3.6-data-science-EXASOL-6.2.0/flavor_base/security_scan/Dockerfile:7]: curl is I think everywhere installed, so maybe use instead curl

**Review by tkilias — CHANGES_REQUESTED** [review]

File: ext/scripts/security_scan/exec_oyster.R [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+library(jsonlite)
+args = commandArgs(trailingOnly=TRUE)
+
+n_vulnerabilities =  nrow(audit_vuln)
+
+report <- toJSON(audit_vuln)
+write(report, paste0(args[1], "/oyster.json"))
+
+exit_code <- 0
+if(n_vulnerabilities > 0) {
```
[the remark this answers is not in the document]
Comment (tomuben) [reply ext/scripts/security_scan/exec_oyster.R:18]: Actually, today on our r-3.5-datascience and r-4-minimal flavors no vulnerability is found (Not even low or medium). So I thought we could keep the rules for r-packages a bit more strict than for the others.

File: flavors/python-3.6-data-science-EXASOL-6.2.0/flavor_base/security_scan/Dockerfile (hunk 2 above)
Comment (tomuben) [reply flavors/python-3.6-data-science-EXASOL-6.2.0/flavor_base/security_scan/Dockerfile:7]: I think this is a good idea

Comment (tomuben) [reply flavors/python-3.6-data-science-EXASOL-6.2.0/flavor_base/security_scan/Dockerfile]: Done

File: ext/scripts/security_scan/exec_oyster.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (tkilias) [reply ext/scripts/security_scan/exec_oyster.R:18]: ok

**Review by tkilias — APPROVED** [review]

**Pull request merged**